### PR TITLE
fix: Server crashes when receiving an array of `Parse.Pointer` in the request body

### LINKED ISSUE #9010: Parse server crash - Cannot read property 'encodeParseObjectInCloudFunction' of undefined in PS6.x.x
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

Parse server crash - Cannot read property 'encodeParseObjectInCloudFunction' of undefined

This issue is the same as https://github.com/parse-community/parse-server/issues/8783#issuecomment-1994944683 but the PR fix for that issue was only pushed to v7.

Opening this new issue for PR for v6

### Steps to reproduce

Same as https://github.com/parse-community/parse-server/issues/8783#issuecomment-1994944683

### Actual Outcome

Parse server crash - Cannot read property 'encodeParseObjectInCloudFunction' of undefined

### Expected Outcome

Not to crash

### Environment

Server
- Parse Server version: `6.5.2`
- Operating system: `Mac`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `local`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `6.0.0`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `Local`

### PR DESCRIPTION
**Pull request by jaysonng**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

Closes: https://github.com/parse-community/parse-server/issues/9010

same as : https://github.com/parse-community/parse-server/issues/8783

## Approach

Copied / exactly the same as PR https://github.com/parse-community/parse-server/pull/8784

## Tasks

- [x] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)
- [ ] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [ ] Add new Parse Error codes to Parse JS SDK

### CODE REVIEW

File: spec/ParseAPI.spec.js [hunk 1]
```diff
@@ -1267,6 +1267,35 @@ describe('miscellaneous', function () {
     });
   });
 
+  it('test cloud function query parameters with array of pointers', done => {
```
Comment (mtrezza) [inline spec/ParseAPI.spec.js]: We don't use chained promises anymore; why is this test not the same as the original fix for PS7?

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by mtrezza** [comment]
Also check lint

File: spec/ParseAPI.spec.js (hunk 1 above)
Comment (jaysonng) [reply spec/ParseAPI.spec.js]: hmm. i copied the one in the fix. maybe i copied an older one? let me double check.

Comment (jaysonng) [reply spec/ParseAPI.spec.js]: I got it from here

https://github.com/parse-community/parse-server/pull/8784/files

Comment (mtrezza) [reply spec/ParseAPI.spec.js]: Yes, the test is different, no?

**Comment by jaysonng** [comment]
> Also check lint

weird. I see below that lint fails because of a white space.

but when I run it locally, I get 

```
> parse-server@6.5.2 lint
> flow && eslint --cache ./

Launching Flow server for /Users/jayson/Projects/Busog/parse-server/server
Spawned flow server (pid=7154)
Logs will go to /private/tmp/flow/zSUserszSjaysonzSProjectszSBusogzSparse-serverzSserver.log
Monitor logs will go to /private/tmp/flow/zSUserszSjaysonzSProjectszSBusogzSparse-serverzSserver.monitor_log
No errors!
```

File: spec/ParseAPI.spec.js [hunk 2]
```diff
@@ -1267,6 +1267,35 @@ describe('miscellaneous', function () {
...
+      const res = response.data.result;
+      expect(res.option).toEqual('1');
+      // Make sure query string params override body params
+      expect(res.other).toEqual('2');
+      expect(res.foo).toEqual('bar');
+      expect(res.arr.length).toEqual(1);
+      done();
+    });
+  });
+
```
Comment (mtrezza) [inline spec/ParseAPI.spec.js]:
```suggestion
 
```

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: spec/ParseAPI.spec.js [hunk 3]
```diff
@@ -1267,6 +1267,35 @@ describe('miscellaneous', function () {
...
+    }).then(response => {
+      const res = response.data.result;
+      expect(res.option).toEqual('1');
+      // Make sure query string params override body params
+      expect(res.other).toEqual('2');
+      expect(res.foo).toEqual('bar');
+      expect(res.arr.length).toEqual(1);
+      done();
+    });
+  });
```
Comment (mtrezza) [inline spec/ParseAPI.spec.js]:
```suggestion
  it('test cloud function query parameters with array of pointers', async () => {
    Parse.Cloud.define('echoParams', req => {
      return req.params;
    });
    const headers = {
      'Content-Type': 'application/json',
      'X-Parse-Application-Id': 'test',
      'X-Parse-Javascript-Key': 'test',
    };
    const response = await request({
      method: 'POST',
      headers: headers,
      url: 'http://localhost:8378/1/functions/echoParams',
      body: '{"arr": [{ "__type": "Pointer", "className": "PointerTest" }]}',
    });
    const res = response.data.result;
    expect(res.arr.length).toEqual(1);
  });
```

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: spec/ParseAPI.spec.js (hunk 1 above)
Comment (mtrezza) [reply spec/ParseAPI.spec.js]: Replaced incorrect test in https://github.com/parse-community/parse-server/pull/9012/commits/0cdf3f30c40cc80dba383fa14bd30a5fdcba4f29

File: spec/ParseAPI.spec.js [hunk 4]
```diff
@@ -1267,6 +1267,25 @@ describe('miscellaneous', function () {
...
+    const response = await request({
+      method: 'POST',
+      headers: headers,
+      url: 'http://localhost:8378/1/functions/echoParams',
+      body: '{"arr": [{ "__type": "Pointer", "className": "PointerTest" }]}',
+    });
+    const res = response.data.result;
+    expect(res.arr.length).toEqual(1);
+  });
+
```
Comment (mtrezza) [inline spec/ParseAPI.spec.js]:
```suggestion

```

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.5.4](https://github.com/parse-community/parse-server/releases/tag/6.5.4)

**Pull request merged**